java/console/src/main/java/Main.java

```diff
@@ -0,0 +1,267 @@
+     * is directly printed to console, without the addition of line breaks.
+     *
+     * @param audioTranscriptDelta The server sent delta containing a new chunk of
+     *                             text transcript data corresponding to
+     *                             the audio file.
+     */
+    private static void consumeAudioTranscriptDelta(ResponseAudioTranscriptDeltaEvent audioTranscriptDelta) {
+        System.out.print(audioTranscriptDelta.getDelta());
+    }
+
+    /**
+     * Callback triggered when the server signals that the audio response transcript
+     * has been completed.
+     */
+    private static void onAudioResponseTranscriptCompleted() {
+        System.out.println("\nAudio transcript complete.");
+    }
+
+    /**
+     * Error handler. We are particularly interested in the
+     * {@link ServerErrorReceivedException} which describes errors
+     * sent by the server as an event, but mapped into an exception for ease of use.
+     *
+     * @param error The error that occurred while consuming the server sent events.
+     */
+    private static void consumeError(Throwable error) {
+        if (error instanceof ServerErrorReceivedException) {
+            ServerErrorReceivedException serverError = (ServerErrorReceivedException) error;
+            RealtimeServerEventErrorError errorDetails = serverError.getErrorDetails();
+            System.out.println("Error type: " + errorDetails.getType());
+            System.out.println("Error code: " + errorDetails.getCode());
+            System.out.println("Error parameter: " + errorDetails.getParam());
+            System.out.println("Error message: " + errorDetails.getMessage());
+        } else {
+            System.out.println(error.getMessage());
+        }
+    }
+
+    /**
+     * Sends the audio file to the Realtime service in consecutive chunks. For async
+     * usage.
+     *
+     * @param client    The Realtime async client.
+     * @param audioFile A representation of the audio file to send
+     * @return A Mono that signals success when it returns {@link Void}, or an error
+     *         otherwise.
+     */
+    public static Mono<Void> sendAudioFileAsync(RealtimeAsyncClient client, AudioFile audioFile) {
+        byte[] audioBytes;
+        try {
+            audioBytes = audioFile.getAudioFilesBytes();
+        } catch (IOException e) {
+            return Mono.error(e);
+        }
+
+        int samplesPerChunk = audioFile.getSampleRate() * audioFile.getChunkDurationMillis() / 1000;
+        int bytesPerChunk = samplesPerChunk * audioFile.getBytesPerSample();
+        for (int i = 0; i < audioBytes.length; i += bytesPerChunk) {
+            int end = Math.min(audioBytes.length, i + bytesPerChunk);
+            byte[] chunk = new byte[end - i];
+            System.arraycopy(audioBytes, i, chunk, 0, end - i);
+            client.sendMessage(new InputAudioBufferAppendEvent(chunk)).block();
+        }
+
+        return Mono.empty();
```

Comment:
There could be a more idiomatic way of signaling completion than this, from a reactor perspective. But that would also be a learning for me :)